genesyscloud/outbound_ruleset/resource_genesyscloud_outbound_ruleset_utils.go

```diff
@@ -0,0 +1,269 @@
+package outbound_ruleset
+
+import (
+	gcloud "terraform-provider-genesyscloud/genesyscloud"
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v105/platformclientv2"
+)
+
+func getOutboundRulesetFromResourceData(d *schema.ResourceData) platformclientv2.Ruleset {
+	name := d.Get("name").(string)
+
+	return platformclientv2.Ruleset{
+		Name:        &name,
+		ContactList: gcloud.BuildSdkDomainEntityRef(d, "contact_list_id"),
+		Queue:       gcloud.BuildSdkDomainEntityRef(d, "queue_id"),
+		Rules:       buildDialerules(d.Get("rules").([]interface{})),
+	}
+}
+
+// buildDialerules maps a []interface{} into a Genesys Cloud *[]platformclientv2.Dialerrule
+func buildDialerules(rules []interface{}) *[]platformclientv2.Dialerrule {
```

Comment:
I was thinking would this be a better naming convention for the flatten and build functions. This function used to be called `buildsdkOutboundRulesetDialerruleSlice`. These names get very long and get even longer depending how many nested resources there are. I also think they're misleading since the function is building a general `Dialerrule` it's not specific to the ruleset. Might save us from creating multiple functions to build the same objects too

Reply:
I am good with that